blocks/core/ff_module/ff_module-task-event/_src/MarkAndGradeTaskEvent.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+var React = require('react');
+var TaskEventBase = require('./TaskEventBase'),
+    taskEventWithOptionalMessage = require('./taskEventWithOptionalMessage');
+
+module.exports.defaultState = React.createClass({
+    displayName: 'MarkAndGradeTaskEventDefault',
+    render: defaultState
+});
+
+module.exports.deletedState = React.createClass({
+    displayName: 'MarkAndGradeTaskEventDeleted',
+    render: taskEventWithOptionalMessage(function(props){
+        return getStatusText('deleted a ', props.event)+ '.';
+    })
+});
+
+function defaultState(){
+    var event = this.props.event,
+        messageText = event.message,
+        markText = getMarkText(event),
+        gradeText = event.grade;
+
+    var statusText = getStatusText(' added a ', event);
+
+    var mark = markText ? <span className="ff_module-task-event__mark">{markText}</span> : null,
+        grade = gradeText ? <span className="ff_module-task-event__grade">{gradeText}</span> : null,
+        sep = (markText && gradeText) ? ', ' : '';
+
+    var markAndGrade = (mark || grade) ? <p className="ff_module-task-event__mark-and-grade">{mark}{sep}{grade}</p> : null;
+
+    var message = messageText ? <p className="ff_module-task-event__message">{messageText}</p> : null;
+
+    return  <TaskEventBase event={event}>
+                <p className="ff_module-task-event__author-action">{event.author.name}{statusText}:</p>
+                {markAndGrade}
+                {message}
+            </TaskEventBase>
+}
+
+function getStatusText(base, event) {
+    switch (true) {
+        case !!(event.mark && !event.grade):
+            return base + 'mark';
+            break;
+        case !!(!event.mark && event.grade):
+            return base + 'grade';
+            break;
+        case !!(event.mark && event.grade):
+            return base + 'mark and grade';
+            break;
```

Comment:
I don't think a switch statement is the best way to handle this. I feel like an `if else` would be much clearer:
```javascript
if (event.mark && !event.grade) {
  return base + 'mark';
} else if (!event.mark && event.grade) {
  return base + 'grade';
} else if (event.mark && event.grade) {
  return base + 'mark and grade';
}
```
Is there a reason that I haven't considered that a `switch` statement might be better? I think:

- `if` is shorter
- `switch` needs the `!!` which is easy to forget
- `if` is a lot more idiomatic. This is a bit fluffy as a reason but I think probably the most important. `switch` statements are not really intended to have a constant expression, where this is exactly what `if else` was designed for.